pkg/clusteragent/languagedetection/patcher.go

```diff
@@ -207,33 +210,50 @@ func (lp *languagePatcher) handleDeploymentEvent(event workloadmeta.Event) {
 		if len(deployment.InjectableLanguages) > 0 {
 			// If some annotations still exist, remove them
 			annotationsPatch := lp.generateAnnotationsPatch(deployment.InjectableLanguages, langUtil.ContainersLanguages{})
-			lp.patchOwner(&owner, annotationsPatch)
-			return
+			err = lp.patchOwner(&owner, annotationsPatch)
+		} else {
+			Patches.Inc(owner.Kind, owner.Name, owner.Namespace, statusSkip)
 		}
-
-		Patches.Inc(owner.Kind, owner.Name, owner.Namespace, statusSkip)
 	} else if event.Type == workloadmeta.EventTypeSet {
 		detectedLanguages := deployment.DetectedLanguages
 		injectableLanguages := deployment.InjectableLanguages
 
 		// Calculate annotations patch
 		annotationsPatch := lp.generateAnnotationsPatch(injectableLanguages, detectedLanguages)
 		if len(annotationsPatch) > 0 {
-			lp.patchOwner(&owner, annotationsPatch)
+			err = lp.patchOwner(&owner, annotationsPatch)
 		} else {
 			Patches.Inc(owner.Kind, owner.Name, owner.Namespace, statusSkip)
 		}
 
 	}
+
+	if err != nil {
+		lp.logger.Errorf("failed to handle deployment event: %v", err)
+	}
+
 }
 
 // patches the owner with the corresponding language annotations
-func (lp *languagePatcher) patchOwner(namespacedOwnerRef *langUtil.NamespacedOwnerReference, annotationsPatch map[string]interface{}) {
+func (lp *languagePatcher) patchOwner(namespacedOwnerRef *langUtil.NamespacedOwnerReference, annotationsPatch map[string]interface{}) error {
+
+	setAnnotations := map[string]string{}
```

Comment:
This makes me think that maybe the function that generates the annotations patch (`generateAnnotationsPatch`) should return a `map[string]string`. Is there a reason why we're not doing that?

Reply:
`generateAnnotationsPatch` function is returning `map[string]string` because in order to remove an annotation from the deployment, we need to associate the annotation key to `nil`, and this is not possible if the return type is `map[string]string`.